async_customerio/client_base.py

```diff
@@ -100,22 +100,45 @@ async def send_request(
         )
         try:
             raw_cio_response: httpx.Response = await self._client.request(
-                method, url, json=json_payload and sanitize(json_payload), headers=merged_headers, auth=auth
+                method,
+                url,
+                json=json_payload and sanitize(json_payload),
+                headers=merged_headers,
+                auth=auth,
+            )
+            raw_cio_response.raise_for_status()
+        except httpx.TransportError as err:
+            # Raise exception alerting user that the system might be
+            # experiencing an outage and refer them to system status page.
+            raise AsyncCustomerIORetryableError(
+                CUSTOMERIO_UNAVAILABLE_MESSAGE.format(klass=type(err), message=err, count=self._retries)
             )
-            result_status = raw_cio_response.status_code
-            if result_status != 200:
-                raise AsyncCustomerIOError(f"{result_status}: {url} {json_payload} {raw_cio_response.text}")
+        except httpx.HTTPStatusError as err:
+            # Check if the status code is retryable
+            if err.response.status_code in [
+                http.HTTPStatus.INTERNAL_SERVER_ERROR,
```

Comment:
In my opinion, 500 code should be removed from the list of retriable codes. It rather means that the backend  experiencing some problem and we do not want to introduce extra load on it.